GVFS/GVFS.Platform.Mac/MacProductUpgraderPlatformStrategy.cs

```diff
@@ -0,0 +1,62 @@
+using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using System;
+using System.IO;
+
+namespace GVFS.Platform.Mac
+{
+    public class MacProductUpgraderPlatformStrategy : ProductUpgraderPlatformStrategy
+    {
+        public MacProductUpgraderPlatformStrategy(PhysicalFileSystem fileSystem, ITracer tracer)
+        : base(fileSystem, tracer)
+        {
+        }
+
+        public override bool TryPrepareLogDirectory(out string error)
+        {
+            error = null;
+            return true;
+        }
+
+        public override bool TryPrepareApplicationDirectory(out string error)
+        {
+            string upgradeApplicationDirectory = ProductUpgraderInfo.GetUpgradeApplicationDirectory();
+
+            Exception deleteDirectoryException;
+            if (this.FileSystem.DirectoryExists(upgradeApplicationDirectory) &&
+                !this.FileSystem.TryDeleteDirectory(upgradeApplicationDirectory, out deleteDirectoryException))
+            {
+                error = $"Failed to delete {upgradeApplicationDirectory} - {deleteDirectoryException.Message}";
+
+                this.TraceException(deleteDirectoryException, nameof(this.TryPrepareApplicationDirectory), $"Error deleting {upgradeApplicationDirectory}.");
+                return false;
+            }
+
+            this.FileSystem.CreateDirectory(upgradeApplicationDirectory);

```

Comment:
@alameenshah, @jamill should we be calling  `TryCreateDirectory` here instead (or catching exceptions that `CreateDirectory` might throw)?

As best I can tell the caller of `TryPrepareApplicationDirectory` does not expect it to throw.